src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -571,28 +579,38 @@ protected static boolean areItemStacksEqual(@NotNull ItemStack stackA, @NotNull
 
     /**
      * Determines if the provided recipe is possible to run from the provided inventory, or if there is anything
-     * preventing
-     * the Recipe from being completed.
+     * preventing the Recipe from being completed.
      * <p>
      * Will consume the inputs of the Recipe if it is possible to run.
      *
-     * @param recipe          - The Recipe that will be consumed from the inputs and ran in the machine
-     * @param importInventory - The inventory that the recipe should be consumed from.
-     *                        Used mainly for Distinct bus implementation for multiblocks to specify
-     *                        a specific bus, or for addons to use external inventories.
-     * @param importFluids    - The tanks that the recipe should be consumed from
-     *                        Used currently in addons to use external tanks.
-     * @return - true if the recipe is successful, false if the recipe is not successful
-     */
-    protected boolean setupAndConsumeRecipeInputs(@NotNull Recipe recipe,
-                                                  @NotNull IItemHandlerModifiable importInventory,
-                                                  @NotNull IMultipleTankHandler importFluids) {
-        this.overclockResults = calculateOverclock(recipe);
-
-        modifyOverclockPost(overclockResults, recipe.getRecipePropertyStorage());
-
-        if (!hasEnoughPower(overclockResults)) {
-            return false;
+     * @param recipe          The Recipe that will be consumed from the inputs and ran in the machine
+     * @param importInventory The inventory that the recipe should be consumed from. Used mainly for Distinct bus
+     *                        implementation for multiblocks to specify a specific bus, or for addons to use external
+     *                        inventories.
+     * @param importFluids    The tanks that the recipe should be consumed from Used currently in addons to use
+     *                        external tanks.
+     * @return the recipe if the setup is successful, null if the setup is not successful
+     */
+    protected final @Nullable Recipe setupAndConsumeRecipeInputs(@NotNull Recipe recipe,
+                                                                 @NotNull IItemHandlerModifiable importInventory,
+                                                                 @NotNull IMultipleTankHandler importFluids) {
+        calculateOverclock(recipe);
+
+        modifyOverclockPost(ocResult, recipe.getRecipePropertyStorage());
+
+        if (ocResult.parallel() > 1) {
+            Recipe r = subTickOC(ocResult, recipe, importInventory, importFluids);
+            if (r == null) {
+                invalidateInputs();
+                return null;
+            }
+
```

Comment:
Why should failing to subTickOC a recipe invalidate the entire recipe? If we fail to subTickOC a recipe, there is still the original recipe that can be run.

Reply:
Parallel logic returns null if the recipe cannot fit in the output space, and it would have found a recipe that fit if it only reached 1 parallel. If a recipe with 1 parallel (not parallelized) cannot fit but earlier on we found that the recipe did fit, then we have something more wrong going on.

`subTickOC` only returns null in cases of failure like the above (or having recipemap be null), so I thought we should just fail in that case. If you think we should fallback to the original recipe, I'll switch over.